### ISSUE TITLE
"nvim -c" does not parse correctly on OS X

### ISSUE THREAD

**Issue by justinmk** [neovim/neovim#1962]
Demonstration:

```
nvim -c 'au VimEnter * Gcommit --amend'
Unknown option argument: "--amend"
```

I can only reproduce this on OS X 10.8.5 (Terminal.app, bash 3.2.53). Does not happen in ubuntu x64 (bash 4.3.11).

The complaint about `--amend` is a red herring. Because without `--amend,` it executes `au` immediately (wrong), and interprets the rest as file arguments (wrong):

```
nvim -u NONE -c 'au VimEnter * echo "foo"'
"VimEnter" [New File]
--- Auto-Commands ---
```

The `*` is expanded and nvim opens all of the corresponding files, as well as creating new buffers for `echo` and `"foo"`.

**Comment by cztchoice** [comment]
Does this still make sense?
I cannot reproduce this in Mac 10.12 (16A323) bash 3.2.57 (or zsh 5.3.1)

When i Execute `nvim -u NONE -c 'au VimEnter * echo "foo"'`, I just got neovim open with foo output on the bottom. This behavior is same with vim 8.0.124

When I execute `nvim -u NONE -c 'au VimEnter * Gcommit --amend'`, I got
```
Error detected while processing VimEnter Auto commands for "*":
E492: Not an editor command: Gcommit --amend
```
The same with vim, too.
